exonum/src/runtime/execution_context.rs

```diff
@@ -269,10 +323,12 @@ impl ExecutionContextUnstable for ExecutionContext<'_> {
             .dispatcher
             .runtime_for_service(instance_id)
             .ok_or(CoreError::IncorrectRuntime)?;
+
         let context = self.child_context(interface_name, descriptor, fallthrough_auth);
         runtime
             .execute(context, method_id, arguments)
             .map_err(|mut err| {
+                self.should_rollback();
                 err.set_runtime_id(runtime_id).set_call_site(|| {
                     CallSite::new(
```

Comment:
Nit: Wouldn't it make sense to use `CallSite::from_call_info` here?

Reply:
I am not have `CallInfo` instance here.